lib/connect.c

```diff
@@ -1075,17 +1075,25 @@ static CURLcode singleipconnect(struct connectdata *conn,
   /* Connect TCP sockets, bind UDP */
   if(!isconnected && (conn->socktype == SOCK_STREAM)) {
     if(conn->bits.tcp_fastopen) {
-#if defined(CONNECT_DATA_IDEMPOTENT) /* OS X */
-      sa_endpoints_t endpoints;
-      endpoints.sae_srcif = 0;
-      endpoints.sae_srcaddr = NULL;
-      endpoints.sae_srcaddrlen = 0;
-      endpoints.sae_dstaddr = &addr.sa_addr;
-      endpoints.sae_dstaddrlen = addr.addrlen;
-
-      rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
-                    CONNECT_RESUME_ON_READ_WRITE | CONNECT_DATA_IDEMPOTENT,
-                    NULL, 0, NULL, NULL);
+#if defined(CONNECT_DATA_IDEMPOTENT) /* Darwin */
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wpartial-availability"
+        if(connectx != NULL) {
+          sa_endpoints_t endpoints;
+          endpoints.sae_srcif = 0;
+          endpoints.sae_srcaddr = NULL;
+          endpoints.sae_srcaddrlen = 0;
+          endpoints.sae_dstaddr = &addr.sa_addr;
+          endpoints.sae_dstaddrlen = addr.addrlen;
+
+          rc = connectx(sockfd, &endpoints, SAE_ASSOCID_ANY,
+                        CONNECT_RESUME_ON_READ_WRITE | CONNECT_DATA_IDEMPOTENT,
+                        NULL, 0, NULL, NULL);
+        }
+        else {
+          return CURLE_NOT_BUILT_IN;
```

Comment:
I think when connectx is called here that `CURLE_NOT_BUILT_IN` should not be returned. Basically it should never get this far that (and can't afaics) that it's called like that, either we should have a DEBUGASSERT there or if the concern is in the future tcp_fastopen could be true by default then better idea use connect as a fallback and replace `return CURLE_NOT_BUILT_IN;` with ` rc = connect(sockfd, &addr.sa_addr, addr.addrlen);`